maintenance/shaka-player.md

```diff
@@ -0,0 +1,185 @@
+# Or if you already have such a branch locally:
+git checkout v3.3.x
+
+# Inspect all commits since a particular main-branch tag.
+# If the most recent v3.3.x release was v3.3.1, the tag
+# v3.3.1-main is the same point in history for the main branch.
+interactive-cherry-pick v3.3.1-main upstream main
+```
+
+Pre-work:
+ 1. Install the interactive cherry-pick tool:
+    https://github.com/shaka-project/shaka-github-tools/tree/main/interactive-cherry-pick#installation
+ 1. Create a local branch in Shaka Player tracking an upstream release branch
+
+Cherry-picking:
+ 1. Cherry-pick the necessary changes **skipping all feature commits** (see also
+    ["What to Pick"](#what-to-pick) below)
+
+    > :information_source: **NOTE**: If you aren't sure about a commit, you can
+    > select it now and change your mind later if there are merge conflicts
+    > and/or test failures
+
+ 1. Fix any merge conflicts that come up, or choose to skip the current commit
+    with `git cherry-pick --skip`
+ 1. Run the compiler with `python build/all.py`
+ 1. Run the tests with `python build/test.py`
+ 1. If any tests fail, make the necessary changes to fix the cherry-picked
+    commit that caused the failure (either to amend the earlier commits, or as
+    a new commit on top)
+ 1. When the build and all tests pass, push the cherry-picked commits upstream
+    with `git push`
+
+
+### What to Pick
+
+ - Most bug fixes are cherry-picked to active release branches.
+ - Some bug fixes won't apply to a branch because they deal with features not
+   present in that branch.
+ - Some bug fixes won't apply to a branch because the code in the branch is too
+   different, in which case [a decision must be made](https://quotecatalog.com/quote/mitch-hedberg-i-write-jokes-f-jpXNBB1)
+   about whether porting the fix to the branch is worth the effort.  The
+   maintainer will have to consider both the severity of the issue and the
+   difficulty of fixing conflicts or reimplementation.
+ - Non-code changes (docs, tests, demo, etc) are sometimes cherry-picked to
+   branches, depending on what the change is and whether it makes sense for
+   that branch.
+ - **Features are never cherry-picked** to release branches, as this would
+   break semantic versioning.
+
+
+## Release Policy
+
+Maintainers can make a feature release at almost any time.  There is no need to
+wait to batch up any particular number of changes or any particular significant
+features.
+
+
+## Branch Maintenance Policy
+
+We will cherry-pick bug fixes to LTS branches, plus the most recent 2 release
+branches.
+
+There are two kinds of branches we will dub LTS:
+ - The last feature release (minor release number) before a breaking release
+   (major release number) will receive fixes for 1 year
+   - _For example, v3.3 (right before v4.0) will be fixed for 1 year past the
+     release of v4.0.0 (until April 30, 2023)._
+ - The release currently used by the Cast Application Framework will continue
```

Comment:
Here it would be good to define whether the maintainers do the cherry-pick or the Google team does it.

Reply:
Yes, definitely.  I think it could reasonably be the responsibility of Googlers to maintain the branch for CAF, since it's not obvious what branch we're on in that closed-source project.

Reply:
Done.